tools/monorepo-utils/src/changelog/lib/github.ts

```diff
@@ -0,0 +1,143 @@
+/**
+ * Internal dependencies
+ */
+import { getPullRequest, isCommunityPullRequest } from '../../core/github/repo';
+import { Logger } from '../../core/logger';
+
+/**
+ * Get relevant data from a pull request.
+ *
+ * @param {Object} options
+ * @param {string} options.owner repository owner.
+ * @param {string} options.name  repository name.
+ * @param {string} prNumber      pull request number.
+ * @return {Promise<object>}     pull request data.
+ */
+export const getPullRequestData = async (
+	options: { owner: string; name: string },
+	prNumber: string
+) => {
+	const { owner, name } = options;
+	const prData = await getPullRequest( { owner, name }, prNumber );
+	const isCommunityPR = isCommunityPullRequest( prData, owner, name );
+	const headOwner = isCommunityPR ? prData.head.repo.owner.login : owner;
+	const branch = prData.head.ref;
+	const fileName = branch.replace( /\//g, '-' );
+	const prBody = prData.body;
+	const head = prData.head.sha;
+	const base = prData.base.sha;
+	return {
+		prBody,
+		isCommunityPR,
+		headOwner,
+		branch,
+		fileName,
+		head,
+		base,
+	};
+};
+
+/**
+ * Determine if a pull request description activates the changelog automation.
+ *
+ * @param {string} body pull request description.
+ * @return {boolean} if the pull request description activates the changelog automation.
+ */
+export const getShouldAutomateChangelog = ( body: string ) => {
+	const regex =
+		/\[x\] Automatically create a changelog entry from the details/gm;
+	return regex.test( body );
+};
+
+/**
+ * Get the changelog significance from a pull request description.
+ *
+ * @param {string} body pull request description.
+ * @return {void|string} changelog significance.
+ */
+export const getChangelogSignificance = ( body: string ) => {
+	const patchRegex = /\[x\] (Patch)\r\n/gms;
```

Comment:
Shouldn't we be able to combine those three expressions into one?

Reply:
Yes, for sure. Do you think the tradeoff on readability is worth the reduction in lines of code?

Reply:
ok, I did this. Should be smaller and readable

```js
const regex = /\[x\] (Patch|Minor|Major)\r\n/gm;
```